### ISSUE TITLE
MapView onViewStateChange leads to React render warning on 8.5.0

### ISSUE THREAD

**Issue by shestakov-bc** [visgl/deck.gl#6024]
This is not observed in 8.4.16, which was our original version before I tried to upgrade.

#### Description
I'm using multiple views, and when setting up main view to control viewState, like this:
const MainView = ({viewState, setViewState}) => {
  return new MapView({
    id: VIEW.MAIN,
    controller: true,
    viewState: viewState,
    onViewStateChange: e => setViewState(e.viewState),
  });
};

Any interaction with map (zoom for example) lead to React warning:
```
Warning: Cannot update a component (`MyMap`) while rendering a different component (`ForwardRef`). To locate the bad setState() call inside `ForwardRef`, follow the stack trace as described in...
```
It's still works, but that warning tells me that something really bad is going on and we can't let it through to release.

#### Expected Behavior
We should be able to use viewState handling on custom Views, not just directly on DeckGL without any react warnings.

#### Repro Steps
```
const views = [MainView({viewState, setViewState}), new OrthographicView({ id: VIEW.OVERLAY })];
...
  return (
    <DeckGL
      // layers
      layers={layers}
      layerFilter={layerFilter}
      // view
      views={views}
      onResize={setViewSize}
      useDevicePixels={false} // to use 1:1 canvas rendering (better performance)
      // controls
      onClick={handleDeckClick}
      getCursor={handleDeckCursor}
      getTooltip={handleDeckTooltip}
      // interaction callables
      onInteractionStateChange={setInteractionState}
      onDragStart={handleDeckDragStart}
      onDrag={handleDeckDrag}
      onDragEnd={handleDeckDragEnd}
    />
  );
```
#### Environment
 - Framework Version: deck.gl 8.5.0
 - Browser Version: Version 1.26.77 Chromium: 91.0.4472.164 (Official Build) (64-bit)
 - OS: Ubuntu 20.04

#### Logs
```
Warning: Cannot update a component (`MyMap`) while rendering a different component (`ForwardRef`). To locate the bad setState() call inside `ForwardRef`, follow the stack trace as described in...
```

**Comment by Pessimistress** [comment]
I cannot reproduce this. Can you provide a CodeSandbox?

**Comment by shestakov-bc** [comment]
sorry for the late response, busy busy busy. I'll try to find time to build one from scratch on weekend. Can't share our work src of course.

**Comment by Pessimistress** [comment]
Closing for inactivity. Reopen if steps to reproduce are provided.

**Comment by Dmi3Dev** [comment]
I apologize that it took me so long to do this. Way too busy at two jobs. Here is the sandbox now using 8.6.0 version of deck which still has this issue. I've added description to readme.md and have a workaround in deck-sandbox.js. Basically you guys made it faster and it is awesome, but React can't keep up with it anymore.
https://github.com/Dmi3Dev/deck-sandbox

**Comment by Pessimistress** [comment]
You are not setting `viewState` correctly. `initialViewState` should only be used if you want to use DeckGL as a stateful component. See https://deck.gl/docs/developer-guide/interactivity#externally-manage-view-state

**Comment by Dmi3Dev** [comment]
Thank you for pointing that out. I now see that these two ways don't produce any warnings:
```
const onViewStateChange = useCallback(({ viewId, viewState }) => {
    if (viewId === VIEW.MAIN) {
      setViewStates(currentViewStates => ({
        [VIEW.MAIN]: viewState,
        [VIEW.HUD]: { ...currentViewStates[VIEW.HUD] },
      }));
    } else if (viewId === VIEW.HUD) {
      setViewStates(currentViewStates => ({
        [VIEW.MAIN]: currentViewStates[VIEW.MAIN],
        [VIEW.HUD]: viewState,
      }));
    }
  }, []);
```
or
```
  const handleViewStateChange = ({ viewId, viewState }) => {
    setViewStates(prev => ({
      ...prev,
      [viewId]: viewState,
    }));
  };
  ```

There are multiple reasons I though it was ok to use `viewState` and `onViewStateChange` directly in the view. View documentation describe to have `viewState` https://deck.gl/docs/api-reference/core/view#viewstate and I believe there was an example showing it being used, but I can't find it anymore. I could also misread one of the examples, so I'm not sure. When I implemented it that way with `8.4.16` it worked perfectly without any warning. I saw the example with managing multiple views in Deck object, but using it directly on the view made much cleaner and easier to read code.

May I suggest adding remark to the documentation to not use `viewState` and `onViewStateChange` directly on the view, so others wouldn't end up in the same situation.

O yeah, and the only reason I was using the `initialViewState` is because of this warning, when using `viewState` in the view directly:
```
 deck: missing `viewState` or `initialViewState`
```

Thanks again.
